lib/pgBackRest/Backup/File.pm

```diff
@@ -174,29 +172,44 @@ sub backupFile
 
             push(
                 @{$rhyFilter},
-                {strClass => BACKUP_FILTER_PAGECHECKSUM,
+                {strClass => "pgBackRest::Backup::Filter::PageChecksum",
                     rxyParam => [$iSegmentNo, $hExtraParam->{iWalId}, $hExtraParam->{iWalOffset}]});
         };
 
         # Add compression
         if ($bCompress)
         {
-            push(@{$rhyFilter}, {strClass => STORAGE_FILTER_GZIP, rxyParam => [{iLevel => $iCompressLevel}]});
+            push(@{$rhyFilter}, {strClass => STORAGE_FILTER_GZIP, rxyParam => [STORAGE_COMPRESS, false, $iCompressLevel]});
+        }
+        # Else add protocol compression if the destination is not compressed and there is no encryption
+        elsif (!defined($strCipherPass))
```

Comment:
So there is a distinction between file compression and protocol compression? The filters look identical except for the compression value passed, so what exactly is the difference?

Reply:
Protocol compression is used when `compress=n`, i.e. when the user wants files in the repo to be uncompressed, perhaps because their filesystem provides compression (e.g. ZFS).

It's still a good idea to have some compression when transferring over a network even if it will be decompressed on the remote side.